src/server/api/routes/users.router.js

```diff
@@ -98,4 +98,64 @@ router.get('/:id/favorites/', (req, res, next) => {
     .catch(next);
 });
 
+/**
+ * @swagger
+ * /user/{ID}/edit-page:
+ *  patch:
+ *    tags:
+ *    - User
+ *    summary: Edit/update user
+ *    description:
+ *      Will edit/update the user's information.
+ *    produces: application/json
+ *    parameters:
+ *      - in: path
+ *        name: ID
+ *        description: ID of the module to patch.
+ *      - in: body
+ *        name: user profile inputs
+ *        description: edit/update the user's information.
+ *        schema:
+ *          type: object
+ *          properties:
+ *            full_name:
+ *               type: string
+ *            email:
+ *               type: string
+ *            address:
+ *               type: string
+ *            zipcode:
+ *               type: string
+ *            city:
+ *               type: string
+ *            country:
+ *               type: string
+ *            mobile:
+ *               type: string
+ *            updated_at:
+ *               type: string
+ *               format: date-time
+ *    responses:
+ *      200:
+ *        description: User was patched
+ *      400:
+ *        description: User Id should be a number
+ *      404:
+ *        description: The user ID you provided does not exist
+ *      5XX:
+ *        description: Unexpected error.
+ */
+router.patch('/:id/edit-page', (req, res, next) => {
+  usersController
+    .editUser(req.params.id, req.body)
+    .then((result) => {
+      if (result === 0) {
```

Comment:
Reading this code, I feel like your error handling is spread out between the router and the controller. 

I would recommend putting those things together in the controller. See my comments in the controller file but I think this code should become simply:
```
  usersController
    .editUser(req.params.id, req.body)
    .then((result) => {
      res.json(result);
    })
    .catch(next);
```

This way your controller stays focused on unpacking the HTTP request, passing the data to the right controller, and then sending back either the response or the error.
There is no other logic here and that is how you want your router to work, the logic lives in the controller (which in turn might talk to a service).

Reply:
I will try to make all the error handeling in the controller.